Fix an issue ALPN handshake test does not respect HTTPS_PROXY

greedy52 opened the pull request:
Fixes #27397 

Related issue

- #27397
- #27461
- #21870

greedy52 committed:
Fix an issue ALPN handshake test does not respect HTTPS_PROXY

marcoandredinis committed:
Docs: improve Postgres in GCP (#27141)

* Docs: improve Postgres in GCP

Add `auth_token` in `teleport.yaml` example.

Show how to set the `GOOGLE_APPLICATION_CREDENTIALS` in a `systemd` env.

* suggest the usage of EnvironmentFile

stevenGravy committed:
include changelog for docs tests (#27449)

* include changelog for docs tests

* correct capitalization for file type

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

* include CHANGELOG.md as ignored in bypass

---------

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

gabrielcorado committed:
MongoDB Atlas IAM authentication docs (#26831)

* docs(db): add section on how to configure mongo atlas aws iam auth

* Apply suggestions from code review

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

* docs(mongodb-atlas): code review changes

---------

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
docs: add enterprise value for kube agent reference (#27446)

* docs: add enterprise value for kube agent reference

* correct spacing

ptgott committed:
Trim yum release version in install-linux.mdx (#27367)

* Trim yum release version in install-linux.mdx

Fixes #20978

The `$VERSION_ID` variable we tell users to use when adding our yum repo
is incorrect, since it does not include a minor version. This
quick-and-dirty fix instructs users to trim `$VERSION_ID` to include
only the major version before adding the repo.

* lint fixes

---------

Co-authored-by: Steven Martin <steven@goteleport.com>

flyinghermit committed:
device trust marshalers (#24963)

* feat: device resource in tctl get all

* check for device resource but ignore in favour of enterprise resource migration

* add firstStart indicator in auth service config. Tests

* remove device bootstrap from this PR

* multiple updates:
- remove resource marshaler, this will be added to e repo instead
- remove device resource checks in oss (not needed as resource marshaler now added to e). tests removed

* move device marshalers to service package

* run fix-imports, fix test

* remove device case from itemsFromResource

GavinFrazar committed:
Document all installer script template vars (#27322)

* Document all installer script template vars

* replace aws ssm reference with generic terms

atburke committed:
Docs: GCP join method (#27054)

* Add GCP join method docs

* Update prerequisites

* Fix linting

* Fix linting

* Address comments

* Fix Kube join link

* Fix punctuation

webvictim commented on api/client/alpn_conn_upgrade.go:
```diff
@@ -48,19 +53,28 @@ import (
-	netDialer := &net.Dialer{
-		Timeout: defaults.DefaultIOTimeout,
-	}
+	// Use NewDailer which takes care of ProxyURL, and use a shorter I/O
```
```suggestion
	// Use NewDialer which takes care of ProxyURL, and use a shorter I/O
```

webvictim commented on api/client/contextdialer.go:
```diff
@@ -48,6 +48,16 @@ type dialConfig struct {
 	// proxyHeaderGetter is used if present to get signed PROXY headers to propagate client's IP.
 	// Used by proxy's web server to make calls on behalf of connected clients.
 	proxyHeaderGetter PROXYHeaderGetter
+	//proxyURLFunc is a function used to get ProxyURL. Defaults to
+	//utils.GetProxyURL if not specified.
```
```suggestion
	// proxyURLFunc is a function used to get ProxyURL. Defaults to
	// utils.GetProxyURL if not specified.
```

Commit:
Converts the default Content-Security-Policy representation to a map (#27182)

* Converts the default Content-Security-Policy representation to a map
which makes it easier to programmatically add-to/overwrite the defaults
for special cases.

Also adds tests for the various custom CSPs.

* Alphabetize CSP directives for ease of testing/debugging in the future

* makes maps more easily composable

* making string concatenation more efficient

* code review tweaks

* plain require.Contains

marcoandredinis committed:
Move Cloud Matchers to proto (#27162)

* Move Cloud Matchers to proto

As described in RFD0125, we are going to create a new resource:
DiscoveryConfig

This new resource will contain the Cloud Matchers as spec fields:
- AWSMatcher
- AzureMatcher
- GCPMatcher

To define grpc methods that use this new resource, we must create all
the matchers types as part of the proto definition.

This PR moves those definitions into types.pb.go and removes them from
`lib/services.{AWS,Azure,GCP}Matcher`.

This should have no side effect in business logic given that we re-use
the same field names everywhere.

* remove most of gogo tags

* Revert "remove most of gogo tags"

This reverts commit 5ecf8d0cec2d7aeb9b79d5c992dfeb8851aea129.

* fix typos

Joerger committed:
Only fallback to SSH_TELEPORT_ env variables for proxy, user, and cluster name when used with headless. (#27057)

stevenGravy committed:
docs: adds configuration and helm chart to app access getting started (#24026)

* initial app access updates

* Add helm and configure for app access start

* Fix app configure

* change to just configure

* Update docs/pages/application-access/getting-started.mdx

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

* fix image description

* verbiage updates

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

* update app access for dns address

* lint fix

---------

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>
Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

Commit:
Docs: Trusted Clusters - Mention the correct expiration time as per tctl command (Buddy PR) (#27285)

* Update trustedclusters.mdx

* Update trustedclusters.mdx

* Update trustedclusters.mdx

---------

Co-authored-by: Puru <5674762+tuladhar@users.noreply.github.com>

Commit:
Update Kubernetes Service Account nav listing... (#27307)

to align with the rest of the items in this sub-menu, as discussed in https://github.com/gravitational/teleport/pull/27054\#pullrequestreview-1456063634

ptgott committed:
Add scaling warning re: DynamoDB (#27369)

Closes #25170

Since we want to keep docs on configuring Teleport's backends within the
Backend Reference, add a Notice to this reference in the DynamoDB
section.

benarent committed:
bump version numbers in Readme (#26387)

benarent committed:
Change Campaign to utm_campaign (#26819)

* Change Campaign to utm_campaign

* yarn prettier-write

* Fix Tests

* Update Support.story.test.tsx.snap

* Update ButtonLockedFeature.test.tsx

michellescripts committed:
add new CTA event property (#27216)

tcsc committed:
RFD: Delivering more secure AMIs (#24726)

* RFD: Delivering more secure AMIs

* Update rfd/XXXX-securing-amis.md

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

* Update rfd/XXXX-securing-amis.md

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

* Update rfd/XXXX-securing-amis.md

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

* Update rfd/XXXX-securing-amis.md

Co-authored-by: Roman Tkachenko <roman@goteleport.com>

* Apply suggestions from code review

Co-authored-by: Roman Tkachenko <roman@goteleport.com>
Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

* Apply suggestions from code review

Co-authored-by: Mike Jensen <jentfoo@users.noreply.github.com>

* US spellings

Co-authored-by: Mike Jensen <jentfoo@users.noreply.github.com>

* US Spelling

Co-authored-by: Mike Jensen <jentfoo@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Gus Luxton <gus@goteleport.com>

* Stash

* Assign number

---------

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>
Co-authored-by: Roman Tkachenko <roman@goteleport.com>
Co-authored-by: Mike Jensen <jentfoo@users.noreply.github.com>
Co-authored-by: Gus Luxton <gus@goteleport.com>

Joerger committed:
Fix an unintended interaction between  and Proxy Templates where the environment variable is prioritized over the template. (#27492)

tcsc committed:
Integrate AMI buids into drone (#27354)

kimlisa committed:
Web: Plugin tweaks and new plugin icons (#27427)

Move plugin kinds so we have only one source of truth

Commit:
Primarily changes "match: '^.*\.dev\.example\.com$'" to "match: '^.*\.dev\.example\.com'" so that users aren't mistakenly guided towards eliminating the implicit ":3389" from their regex matches (#27423)

nklaassen committed:
fix: record applied login rules in github login event (#27437)

jakule committed:
Use Ubuntu 22.04 for the buildbox (#27497)

After moving the Connect to a separate Docker image https://github.com/gravitational/teleport/pull/27175 we're able to use the latest ubuntu LTS on our build image. We're not using this image to produce any releases (only CI runs), so updating the image will have no effect on our releases.

camscale committed:
Fix build for new buildbox (#27521)

* Revert the removal of quintush/helm-unittest

Commit 5d53c91c7a8d41391422171749b3a3d7f23084ab removed
quintush/helm-unittest from the buildbox. It seems we still need that
version so revert those hunks that removed it.

* tests: Fix invalid sudoers file test

Fix the invalid sudoers file test to not look for an exact string but
just a substring. The error message has changed from Ubuntu 20.04 to
22.04 and it has removed some extra wording.

stevenGravy committed:
docs: include token generation for app access getting started (#27511)

camscale committed:
tooling: Add a flaky test report to render-tests (#27098)

* tooling: Refactor render-tests

Refactor render-tests to simplify and group the logic prior to adding a
mode to report flaky tests. No additional functionality has been added,
but the coverage detection has been fixed as the regexp was incorrect
(presumably the output format changed).

* tooling: Rework render-tests counting

Rework the counting of pass/fail/skip events in render-test to record
these counts at the test and package level instead of a single status of
pass/fail/skip. This will allow us to accumulate results from more than
one test run to be able to test for flaky tests.

Tally and output package and test counts separately as previously it was
technically incorrect when it said "n tests passed", as that also
included package results. This may make it a little clearer just how
much is failing when a failure occurs.

* tooling: Add flakiness mode to render-tests

Add a `-report-by flakiness` mode to `render-tests` that can accumulate
multiple test runs and report the top N flaky tests. This mode is
intended to be used in a daily run of the tests for a couple of hours,
or perhaps 200 times, and to report on the tests that have failed the
most.

The output of multiple test runs should be fed into the single run of
`render-tests`. A `rerun` utility is forthcoming with which you could
do:

    rerun -n 200 -t 2h go test -shuffle on -cover -json . | \
        render-tests -report-by flakiness -top 10

* tooling: Have render-tests write summary to file

Add the `-summary-file` flag to have render-tests write a summary of the
test run to the specified file. This is to be used to get a flaky test
summary that can be sent on slack via CI (GitHub Actions).

* tooling: Add rerun command for multiple test runs

Add a `rerun` command that is intended to be used to run tests multiple
times for a duration. It allows `go test -json` to be run many times
with the output piped to `render-tests -report-by flakiness` to generate
a summary of flaky tests over a large number of runs.

kimlisa committed:
Update e (#27523)

stevenGravy committed:
Update description of Roles UI (#22404)

* Update description of Roles UI

* Update phrasing

* lint fix

* Update role description

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

* Change verbiage

* lint fix

---------

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

public-renovate-gha[bot] committed:
Configure Renovate (#27410)

* Add renovate.json

* move renovate.json

* add renovate workflow

* update subset of dependencies for now

* sort ignores alphabetically

* fix: add ignored dependencies for Teleport Go deps

* sort ignored deps

* fix: use excludePackageNames instead of exludeDepNames

* disable certain packages

* ignoring replaced dependencies

* remove on pull request from workflow

---------

Co-authored-by: public-renovate-gha[bot] <135069952+public-renovate-gha[bot]@users.noreply.github.com>
Co-authored-by: Tim Ross <tim.ross@goteleport.com>

stevenGravy committed:
use proxy port in openssh config (#27536)

* use proxy port in openssh config

* update test record

jakule committed:
Implement in-memory vector DB (#27329)

In order to include embeddings in the Assist conversations, we need a way to store and retrieve when needed. This PR adds vector store/search implementation based on k-d tree.

public-renovate-gha[bot] committed:
chore(deps): update assets - backport (#27543)

Co-authored-by: public-renovate-gha[bot] <135069952+public-renovate-gha[bot]@users.noreply.github.com>

public-renovate-gha[bot] committed:
chore(deps): update build assets - tooling (#27542)

Co-authored-by: public-renovate-gha[bot] <135069952+public-renovate-gha[bot]@users.noreply.github.com>

ptgott committed:
Fix redirects (#27488)

The `/docs/cloud` redirect was 404ing because
`/docs/choose-an-edition/teleport-cloud/` does not correspond to a page
within the docs. This updates the redirects for this and similar
configurations.

jentfoo committed:
Upgrade mongodb to v1.11.7 (#27429)

* Upgrade mongodb to v1.11.6

Includes some minor code changes around how the mongodb topology is configured.
This drops support for MongoDB prior to version 3.6.  For that reason unit tests don't currently pass.

* access_test: Update MongoDB supported wire versions

Removes support for MongoDB prior to version 3.6

* Fixes after rebase (more needed changes for auth implementation)

* Upgrade mongo to just release 1.11.7 tag

TeleLos committed:
Update README.md (#22468)

updated minor typo.

wadells committed:
Fix the default `teleport-kube-agent` upgrade server (#27504)

updates.gravitational.io does not exist, but
`updates.releases.teleport.dev does, and is the correct endpoint.

public-renovate-gha[bot] committed:
chore(deps): update rdp client (#27558)

Co-authored-by: public-renovate-gha[bot] <135069952+public-renovate-gha[bot]@users.noreply.github.com>

yjperez committed:
Update to Readme for Teleport Usage (#24899)

* Update to Readme for Teleport Usage

Cleaning up the Readme. Removing the prompt option as it is no longer an option. Also clarifying where to find the container image version. Lastly, reordered the docker command to be backwards compatible on Docker.

* Update examples/teleport-usage/README.md

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

* Update examples/teleport-usage/README.md

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

---------

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

Commit:
Update host user creation RFD to  allow for permanent users (#27074)

* Update rfd-0057 (host user creation) to allow permanent users

* Rename to keep and use keep by default

* Include deprecation migration plan

greedy52 commented:
seems github is not happy with the rebase. closing this and will open a new one.
